src/main/java/seedu/address/logic/commands/delivery/DeliveryFindCommand.java

```diff
@@ -0,0 +1,67 @@
+package seedu.address.logic.commands.delivery;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.Messages.MESSAGE_USER_NOT_AUTHENTICATED;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.delivery.DeliveryNameContainsKeywordsPredicate;
+
+/**
+ * Finds and lists all deliveries in delivery book whose name contains any of the argument keywords.
+ * Keyword matching is case insensitive.
+ */
+public class DeliveryFindCommand extends Command {
+
+    public static final String COMMAND_WORD = "delivery find";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all deliveries whose names contain any of "
+        + "the specified keywords (case-insensitive) and displays them as a list with ID numbers.\n"
+        + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
+        + "Example: " + COMMAND_WORD + " chocolate vanilla";
+
+    private final DeliveryNameContainsKeywordsPredicate predicate;
+
+    public DeliveryFindCommand(DeliveryNameContainsKeywordsPredicate predicate) {
+        this.predicate = predicate;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        // User cannot perform this operation before logging in
+        if (!model.getUserLoginStatus()) {
+            throw new CommandException(MESSAGE_USER_NOT_AUTHENTICATED);
+        }
+
+        model.updateFilteredDeliveryList(predicate);
+        return new CommandResult(
+            String.format(Messages.MESSAGE_DELIVERY_LISTED_OVERVIEW,
+                model.getFilteredDeliveryList().size()), true);
```

Comment:
Should we do another getter method for this under `model`? 🤔

Reply:
I think we should, as this might violate LoD, creating a new issue for consideration